warehouse/ingest-core/src/main/java/datawave/ingest/mapreduce/job/reindex/ShardReindexJob.java

```diff
@@ -0,0 +1,846 @@
+        @Parameter(names = "--startDate", description = "yyyyMMdd start date")
+        private String startDate;
+
+        @Parameter(names = "--endDate", description = "yyyyMMdd end date")
+        private String endDate;
+
+        @Parameter(names = "--splitsPerDay", description = "splits for each day")
+        private int splitsPerDay;
+
+        @Parameter(names = "--table", description = "shard table")
+        private String table = "shard";
+
+        @Parameter(names = "--accumuloMetadata", description = "fetch files from the accumulo.metadata table for the given start/end dates")
+        private boolean accumuloMetadata = false;
+
+        @Parameter(names = "--blocksPerSplit", description = "Number of rfile index blocks per split, -1 to not split rfiles")
+        private int blocksPerSplit = -1;
+
+        @Parameter(names = "--useScanServers", description = "Use scan servers for any accumulo scans")
+        private boolean useScanServers = false;
+
+        @Parameter(names = "--accumuloLocal", description = "Run the accumulo iterators local in offline mode")
+        private boolean accumuloLocal = false;
+
+        @Parameter(names = "--accumuloClientPropertiesPath", description = "Filesystem path to accumulo-client.properties file to use for the accumulo client")
+        private String accumuloClientPropertiesPath;
+
+        // alternatively accept RFileInputFormat
+        @Parameter(names = "--inputFiles", description = "When set these files will be used for the job. Should be comma delimited hdfs glob strings",
+                        required = false)
+        private String inputFiles;
+
+        @Parameter(names = "--sourceHdfs", description = "HDFS for --inputFiles", required = true)
+        private String sourceHdfs;
+
+        // support for cache jars
+        @Parameter(names = "--cacheDir", description = "HDFS path to cache directory", required = true)
+        private String cacheDir;
+
+        @Parameter(names = "--cacheJars", description = "jars located in the cacheDir to add to the classpath and distributed cache", required = true)
+        private String cacheJars;
+
+        // work dir
+        @Parameter(names = "--workDir", description = "Temporary work location in hdfs", required = true)
+        private String workDir;
+
+        // support for additional resources
+        @Parameter(names = "--resources", description = "configuration resources to be added")
+        private String resources;
+
+        @Parameter(names = "--reducers", description = "number of reducers to use", required = true)
+        private int reducers;
+
+        @Parameter(names = "--outputDir", description = "output directory that must not already exist", required = true)
+        private String outputDir;
+
+        @Parameter(names = "--destHdfs", description = "HDFS for --outputDir", required = true)
+        private String destHdfs;
```

Comment:
In theory a separate `destHdfs` shouldn't be needed if `outputDir` is a fully qualified path like `hdfs://ingest/foo/bar` and you use something like `path.getFilesystem(conf)` when you need to get the `FileSystem` object.